### PR TITLE
perf_tests: remove false dependence on google test

### PR DESCRIPTION
**Pull request by cwpearson**
Lots of stuff that did and didn't need gtests was in `KokkosKernels_TestUtils.hpp`, so I broke some of that functionality out into new utility files.

perf_tests now includes those new utility files and not need to drag google test in without using it.

This will make it a bit easier to update gtest to a new version.

### CODE REVIEW

File: test_common/KokkosKernels_TestVanilla.hpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const typename Kokkos::TeamPolicy<ExecutionSpace>::member_type& team) const {
+    int i = team.league_rank();
+    SubviewTypeA _A;
+    SubviewTypeB _B;
+    SubviewTypeC _C;
+
+    if (batch_size_last_dim) {
+      _A = Kokkos::subview(A, Kokkos::ALL(), Kokkos::ALL(), i);
```
Comment (lucbv) [inline test_common/KokkosKernels_TestVanilla.hpp:101]: Leading underscore with a capital letter...

File: perf_test/sparse/CMakeLists.txt [hunk 2]
```diff
@@ -103,7 +103,6 @@ KOKKOSKERNELS_ADD_EXECUTABLE(
 KOKKOSKERNELS_ADD_EXECUTABLE(
         sparse_gs
         SOURCES KokkosSparse_gs.cpp
-        TESTONLYLIBS kokkoskernelsperf_gtest
```
Comment (lucbv) [inline perf_test/sparse/CMakeLists.txt:106]: What's that change about?

File: perf_test/blas/blas3/KokkosBlas3_gemm_perf_test.hpp [hunk 3]
```diff
@@ -28,8 +28,10 @@
 #include "KokkosBatched_HostLevel_Gemm.hpp"
 #include "KokkosBatched_Gemm_Decl.hpp"
 #include "KokkosBatched_Util.hpp"
-#include "gtest/gtest.h"  // EXPECT_NEAR
-#include "KokkosKernels_TestUtils.hpp"
+// #include "gtest/gtest.h"  // EXPECT_NEAR
```
Comment (lucbv) [inline perf_test/blas/blas3/KokkosBlas3_gemm_perf_test.hpp:31]: Just remove it if not needed, we don't want someone to uncomment it later.

**Review by lucbv — CHANGES_REQUESTED** [review]
I have a question and a couple of changes would be cleaner

File: test_common/KokkosKernels_TestVanilla.hpp (hunk 1 above)
Comment (cwpearson) [reply test_common/KokkosKernels_TestVanilla.hpp:101]: I'll make a pass on all of these at once to fix in a future PR

File: perf_test/sparse/CMakeLists.txt (hunk 2 above)
Comment (cwpearson) [reply perf_test/sparse/CMakeLists.txt:106]: This linked against the gtest library even though it didn't need to.

File: test_common/KokkosKernels_TestEpsilon.hpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+#ifndef KOKKOSKERNELS_EPSILON_HPP
+#define KOKKOSKERNELS_EPSILON_HPP
+
+#include <limits>
+
+namespace Test {
+
+template <class T>
+class epsilon {
```
[the remark this answers is not in the document]
Comment (cwpearson) [reply test_common/KokkosKernels_TestEpsilon.hpp:25]: Probably, let me try.

[the remark this answers is not in the document]
Comment (cwpearson) [reply test_common/KokkosKernels_TestEpsilon.hpp:25]: Turns out this was only used in the GEMM perf test.

**Review by lucbv — APPROVED** [review]

**Pull request merged**